flink-core/src/main/java/org/apache/flink/api/connector/sink2/Sink.java

```diff
@@ -52,11 +54,30 @@ public interface Sink<InputT> extends Serializable {
      * @param context the runtime context.
      * @return A sink writer.
      * @throws IOException for any failure during creation.
+     * @deprecated Please implement {@link #createWriter(WriterInitContext)}. For backward
+     *     compatibility reasons - to keep {@link Sink} a functional interface - Flink did not
+     *     provide a default implementation. New {@link Sink} implementations should implement this
+     *     method, but it will not be used, and it will be removed in 1.20.0 release. Do not use
+     *     {@link Override} annotation when implementing this method, to prevent compilation errors
+     *     when migrating to 1.20.x release.
      */
+    @Deprecated
```

Comment:
do we use the attribute option `since` in other places? it might be useful to have that in the annotation itself

Reply:
`since` is only supported since java9. The code should be compatible with java8, so we can not use `since` yet.